cypress/integration/preference.spec.ts

```diff
@@ -1,26 +1,261 @@
+    checkLang('de', 'Sprache', 'Ohne Titel');
+  });
+
+  it('choose en and see if the translation of the preference page gets changed', () => {
+    go2Preference();
+    checkLang('en', 'Language', 'Untitled');
+  });
+
+  it('choose es and see if the translation of the preference page gets changed', () => {
+    go2Preference();
+    checkLang('es', 'Idioma', 'Sin título');
+  });
+
+  it('choose ja and see if the translation of the preference page gets changed', () => {
+    go2Preference();
+    checkLang('ja', '言語', '無題');
   });
 
   it('choose zh-tw and see if the translation of the preference page gets changed', () => {
-    cy.url().should('contain', '#/studio/settings');
-    cy.get('select#select-lang').select('zh-tw');
-    cy.get('div.subtitle:first-child').should('have.text', '一般');
+    go2Preference();
+    checkLang('zh-tw', '語言', '無標題');
+  });
+
+  it('choose zh-cn and see if the translation of the preference page gets changed', () => {
+    go2Preference();
+    checkLang('zh-cn', '语言', '无标题');
+  });
+
+  it('change units and see if home page gets changed ', () => {
+    go2Preference();
+    cy.get('#set-default-units').select('inches');
+    cy.get('#preview-input').should('have.attr', 'value', '3.94');
+    cy.get('#diode-preview-input').should('have.attr', 'value', '2.36');
+    cy.contains('in/s').should('exist');
+    cy.contains('in').should('exist');
     cy.get('div.btn-done').click();
+    cy.get('#speed').should('have.attr', 'value', '0.79');
+    cy.contains('in/s').should('exist');
   });
 
-  it('exit the preference page and check the translation', () => {
-    cy.url().should('contain', '#/studio/beambox');
-    cy.get('div.modal-body').should('exist');
-    cy.get('button[data-test-key="否"]').click();
-    cy.get('div.top-bar > div.file-title').should('have.text', '無標題');
+  it('change font and see if home page gets changed ', () => {
+    go2Preference();
+    cy.get('#set-default-font-family').select('Airstream NF');
+    cy.get('div.btn-done').click();
+    cy.get('div#left-Text>img').click();
+    cy.get('svg#svgcontent').realClick({ x: 100, y: 200 }).realType('Bring Any Design to Life');
+    cy.get('.react-select__value-container').should('have.text', 'Airstream NF');
   });
+
+  it('change Font Style and see if home page gets changed ', () => {
+    go2Preference();
+    cy.get('#set-default-font-style').select('Bold');
+    cy.get('div.btn-done').click();
+    cy.get('div#left-Text>img').click();
+    cy.get('svg#svgcontent').realClick({ x: 100, y: 200 }).realType('Bring Any Design to Life');
+    cy.get('.select-container > select').find('option:selected').should('have.text', 'Bold');
+  });
+
+  it('change document setting and see if home page gets changed ', () => {
+    go2Preference();
+    cy.get('#set-default-model').select('fbm1');
+    cy.get('div.btn-done').click();
+    cy.get('#canvasBackground').should(($value) => { expect($value.attr('width')).to.be.closeTo(672, 2); });
```

Comment:
use `#svgroot` for a more precise result.

Reply:
finished